app/src/main/java/io/gnosis/safe/ui/transactions/details/ConfirmRejectionFragment.kt

```diff
@@ -39,13 +56,49 @@ class ConfirmRejectionFragment : BaseViewBindingFragment<FragmentConfirmRejectio
                 underline = true
             )
             confirmRejection.setOnClickListener {
-                findNavController().navigateUp()
-                findNavController().currentBackStackEntry?.savedStateHandle?.set(REJECTION_CONFIRMATION_RESULT, true)
+                confirmRejection.isEnabled = false
+                viewModel.submitRejection()
             }
         }
-    }
+        binding.confirmRejection.isEnabled = false
+        viewModel.state.observe(viewLifecycleOwner, Observer { state ->
+            when (val viewAction = state.viewAction) {
+                is RejectionSubmitted -> {
+                    Navigation.findNavController(view).navigate(
+                        ConfirmRejectionFragmentDirections.actionConfirmRejectionFragmentToTransactionsFragment(
+                            TxPagerAdapter.Tabs.QUEUE.ordinal
+                        )
+                    )
+                    findNavController().currentBackStackEntry?.savedStateHandle?.set(REJECTION_CONFIRMATION_RESULT, true)
+                }
+                is BaseStateViewModel.ViewAction.Loading -> {
+                    if (!viewAction.isLoading) {
+                        binding.confirmRejection.isEnabled = true
+                    }
+                }
+                is BaseStateViewModel.ViewAction.ShowError -> {
+                    binding.confirmRejection.isEnabled = true
 
-    companion object {
-        const val REJECTION_CONFIRMATION_RESULT = "args.boolean.rejection_confirmation_result"
+                    viewAction.error.let {
+                        when (it) {
+                            is TxRejectionFailed -> {
+                                val error = it.cause.toError()
+                                if (error.trackingRequired) {
+                                    tracker.logException(it.cause)
+                                }
+                                errorSnackbar(requireView(), error.message(requireContext(), R.string.error_description_tx_rejection))
+                            }
+                            else -> {
+                                val error = it.toError()
```

Comment:
```val error = it.cause.toError()
                                if (error.trackingRequired) {
                                    tracker.logException(it.cause)
                                }
```
this part is same for all branches and can be moved outside when

Reply:
For `TxRejectionFailed` error is in `cause` while for the other two it is the Throwable itself. Isn't that enough of a difference to have when  or if-else?

Reply:
What do you think about changing `logException()` to take a Boolean parameter and always pass `error.trackingRequired`. 

If that's ok for you, I'd use it everywhere.

Reply:
I think logException should do what it says - log exception without additional checks

Reply:
can we then propagate cause directly in TxRejectionFailed, so we don't have this extra logic in fragment?

Reply:
You mean `cause` instead of `TxRejectionFailed`? Is that a good idea? It is not that much logic, is it?

Reply:
no, I mean passing cause directly to TxRejectionFailed as a parameter

Reply:
Like this? ef0e73fe4e3a6608c9cd6984327a2467ea5881dd